File: tutorial/sort/insertion/is.py
def Insertion(list):
    for i in range(len(list)):
        key = list[i]
        j = i-1
        while j >= 0 and key < list[j]:
            list[j+1] = list[j]
            j -= 1

        list[j+1] = key


list = [64, 34, 25, 12, 22, 11, 90]
Insertion(list)
for i in list:
    print(i)

File: tutorial/sort/insertion/is.go
package main

import "fmt"

func Insertion(arr []int) {
	for i := 0; i < len(arr); i++ {
		key := arr[i]
		j := i - 1
		for j >= 0 && key < arr[j] {
			arr[j+1] = arr[j]
			j--

		}
		arr[j+1] = key

	}
}
func main() {
	arr := []int{7, 9, 2, 5, 3, 1}
	Insertion(arr)
	for i := 0; i < len(arr); i++ {
		fmt.Println(arr[i])
	}
}
